boost-common/src/main/java/io/openliberty/boost/common/utils/SpringBootUtil.java

```diff
@@ -111,6 +111,44 @@ public static File copySpringBootUberJar(File artifact, BoostLoggerI logger) thr
         return null;
     }
 
+    /**
+     * Copy the Spring Boot Uber JAR to a .spring extension to preserve it
+     * 
+     * @param springBootUberJar
+     * @param projectArtifact
+     * @param shouldReplaceProjectArchive
+     * @param boostLogger
+     * @throws BoostException
+     */
+
+    public static void copySpringBootUberJar(File springBootUberJar, File projectArtifact, boolean shouldReplaceProjectArchive, BoostLoggerI boostLogger) throws BoostException {
+        try {
+            File springBootUberJarCopy = null;
+            if (springBootUberJar != null) { // Only copy the Uber JAR if it is valid
+                springBootUberJarCopy = SpringBootUtil.copySpringBootUberJarFile(springBootUberJar,
+                        boostLogger);
+            }
+
+            if (springBootUberJarCopy == null) {
+                boostLogger.info("Plugin should replace the project archive: " + shouldReplaceProjectArchive);
+                if (shouldReplaceProjectArchive) {
+                    if (projectArtifact != null) {
+                        File springJar = new File(
+                            getBoostedSpringBootUberJarPath(projectArtifact));
+                        if (net.wasdev.wlp.common.plugins.util.SpringBootUtil.isSpringBootUberJar(springJar)) {
+                            boostLogger.info("Copying back Spring Boot Uber JAR as project artifact.");
+                            FileUtils.copyFile(springJar, projectArtifact);
+                        }
+                    }
+                }
+            } else {
+                boostLogger.info("Copied Spring Boot Uber JAR to " + springBootUberJarCopy.getCanonicalPath());
+            }
+        } catch (Exception e) {
+            throw new BoostException(e.getMessage(), e);
```

Comment:
Not very useful here to just propagate the same error message. I would have the same thing here as in line 108:

```
throw new BoostException("Error copying Spring Boot Uber JAR.", e);
```